plugins/mapsmenu.sma

```diff
@@ -14,499 +14,612 @@
-			{
-				message_begin(MSG_ALL, SVC_INTERMISSION)
-				message_end()
+			set_task(10.0, "autoRefuse", 4545454);
+		} 
+		else 
+		{
+			new _modName[10], tempMap[32];		
+			get_modname(_modName, charsmax(_modName));	
+					
+			if (!equal(_modName, "zp"))		
+			{		
+				message_begin(MSG_ALL, SVC_INTERMISSION);
+				message_end();
 			}
-			new tempMap[32];
+			
 			ArrayGetString(g_mapName, g_choosed, tempMap, charsmax(tempMap));
-			set_task(2.0, "delayedChange", 0, tempMap, strlen(tempMap) + 1)
+			set_task(2.0, "delayedChange", 0, tempMap, strlen(tempMap) + 1);
 		}
-	} else {
-		client_print(0, print_chat, "%L", LANG_PLAYER, "VOTE_FAILED")
-		log_amx("Vote: %L", "en", "VOTE_FAILED")
+	} 
+	else 
+	{
+		client_print(0, print_chat, "%l", "VOTE_FAILED");
+		log_amx("Vote: %L", "en", "VOTE_FAILED");
 	}
 	
-	remove_task(34567 + id)
+	remove_task(34567 + id);
 }
 
-public voteCount(id, key)
+public voteCount(id, Menu, Item)
 {
-	if (key > 3)
+	if (Item < 0)
 	{
-		client_print(0, print_chat, "%L", LANG_PLAYER, "VOT_CANC")
-		remove_task(34567 + id)
-		set_cvar_float("amx_last_voting", get_gametime())
-		log_amx("Vote: Cancel vote session")
+		return PLUGIN_HANDLED;
+	}
+	if (Item > 3)
+	{
+		client_print(0, print_chat, "%l", "VOT_CANC");
+		remove_task(34567 + id);
+		set_cvar_float("amx_last_voting", get_gametime());
+		log_amx("Vote: Cancel vote session");
 		
-		return PLUGIN_HANDLED
+		return PLUGIN_HANDLED;
 	}
 	
 	if (get_cvar_float("amx_vote_answers"))
 	{
-		new name[MAX_NAME_LENGTH]
+		new name[MAX_NAME_LENGTH], MapName[32], Access, Info[1], Callback;
 		
-		get_user_name(id, name, charsmax(name))
-		client_print(0, print_chat, "%L", LANG_PLAYER, "X_VOTED_FOR", name, key + 1)
+		get_user_name(id, name, charsmax(name));
+		menu_item_getinfo(Menu, Item, Access, Info, charsmax(Info), MapName, charsmax(MapName), Callback);
+		client_print(0, print_chat, "%l", "X_VOTED_FOR", name, MapName);
 	}
 	
-	++g_voteCount[key]
+	++g_voteCount[Item];
 	
-	return PLUGIN_HANDLED
+	return PLUGIN_HANDLED;
 }
 
-isMapSelected(id, pos)
+bool:isMapSelected(id, pos)
 {
-	for (new a = 0; a < g_voteSelectedNum[id]; ++a)
-		if (g_voteSelected[id][a] == pos)
-			return 1
-	return 0
+	for (new i = 0; i < g_voteSelectedNum[id]; ++i)
+		if (g_voteSelected[id][i] == pos)
+			return true;
+	return false;
 }
 
-displayVoteMapsMenu(id, pos)
+displayVoteMapsMenu(id, Page = 0, Search[] = "")
 {
-	if (pos < 0)
-		return
-
-	new menuBody[512], b = 0, start = pos * 7
-
-	if (start >= g_mapNums)
-		start = pos = g_menuPosition[id] = 0
-
-	new len = format(menuBody, charsmax(menuBody), g_coloredMenus ? "\y%L\R%d/%d^n\w^n" : "%L %d/%d^n^n", id, "VOTEMAP_MENU", pos + 1, (g_mapNums / 7 + ((g_mapNums % 7) ? 1 : 0)))
-	new end = start + 7, keys = MENU_KEY_0
-
-	if (end > g_mapNums)
-		end = g_mapNums
-
-	new tempMap[32];
-	for (new a = start; a < end; ++a)
+	new menuBody[64], tempMap[32];
+	formatex(menuBody, charsmax(menuBody), "\y%l", "CHANGLE_MENU");
```

Comment:
formatex(menuBody, charsmax(menuBody), "\y%L", LANG_PLAYER, "VOTEMAP_MENU");

The menu name is displayed incorrectly